sched/semaphore/sem_destroy.c

```diff
@@ -74,9 +74,9 @@ int nxsem_destroy(FAR sem_t *sem)
    * leave the count unchanged but still return OK.
    */
 
-  old = atomic_read(NXSEM_COUNT(sem));
   do
     {
+      old = atomic_read(NXSEM_COUNT(sem));
```

Comment:
but atomic_try_cmpxchg_release at line 85 will update old to the latest value, why do we need reread it here?

Reply:
ah shoot. you are right. closing this.